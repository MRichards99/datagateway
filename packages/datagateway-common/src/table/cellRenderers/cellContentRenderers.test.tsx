import React from 'react';
import { createShallow } from '@material-ui/core/test-utils';
import { MemoryRouter } from 'react-router-dom';
import { Link } from '@material-ui/core';
import {
  formatBytes,
  datasetLink,
  investigationLink,
} from './cellContentRenderers';

describe('Cell content renderers', () => {
  let shallow;

  beforeEach(() => {
    shallow = createShallow({});
  });

  describe('formatBytes', () => {
    it('converts to bytes correctly', () => {
      expect(formatBytes(10000)).toEqual('9.77 KB');
    });

    it('handles 0 correctly', () => {
      expect(formatBytes(0)).toEqual('0 B');
    });
  });

  describe('datasetLink', () => {
    it('renders correctly', () => {
      const wrapper = shallow(
        <MemoryRouter>{datasetLink('1', 2, 'test')}</MemoryRouter>
      );
      expect(wrapper.find(Link)).toMatchSnapshot();
    });
  });

  describe('investigationLink', () => {
    it('renders correctly', () => {
      const wrapper = shallow(
        <MemoryRouter>{investigationLink(1, 'test')}</MemoryRouter>
      );
      expect(wrapper.find(Link)).toMatchSnapshot();
    });
  });
});
