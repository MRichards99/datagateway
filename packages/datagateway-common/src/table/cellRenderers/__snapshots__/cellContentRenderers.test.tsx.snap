// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cell content renderers datasetLink renders correctly 1`] = `
<WithStyles(ForwardRef(Link))
  component={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "displayName": "Link",
      "propTypes": Object {
        "innerRef": [Function],
        "onClick": [Function],
        "replace": [Function],
        "target": [Function],
        "to": [Function],
      },
      "render": [Function],
    }
  }
  to="/browse/investigation/1/dataset/2/datafile"
>
  test
</WithStyles(ForwardRef(Link))>
`;

exports[`Cell content renderers investigationLink renders correctly 1`] = `
<WithStyles(ForwardRef(Link))
  component={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "displayName": "Link",
      "propTypes": Object {
        "innerRef": [Function],
        "onClick": [Function],
        "replace": [Function],
        "target": [Function],
        "to": [Function],
      },
      "render": [Function],
    }
  }
  to="/browse/investigation/1/dataset"
>
  test
</WithStyles(ForwardRef(Link))>
`;

exports[`Cell content renderers tableLink renders correctly 1`] = `
<WithStyles(ForwardRef(Link))
  component={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "displayName": "Link",
      "propTypes": Object {
        "innerRef": [Function],
        "onClick": [Function],
        "replace": [Function],
        "target": [Function],
        "to": [Function],
      },
      "render": [Function],
    }
  }
  to="/test/url"
>
  test text
</WithStyles(ForwardRef(Link))>
`;
